cmd/k8s-proxy/main.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	kzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"tailscale.com/hostinfo"
+	"tailscale.com/ipn/store/kubestore"
+	apiproxy "tailscale.com/k8s-operator/api-proxy"
+	"tailscale.com/kube/kubetypes"
+	"tailscale.com/tsnet"
+	"tailscale.com/types/logger"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	var (
+		podName = os.Getenv("POD_NAME")
+	)
+
+	if podName == "" {
+		return fmt.Errorf("POD_NAME environment variable is not set")
+	}
+
+	var opts []kzap.Opts
+	switch "dev" { // TODO(tomhjp): make configurable
+	case "info":
+		opts = append(opts, kzap.Level(zapcore.InfoLevel))
+	case "debug":
+		opts = append(opts, kzap.Level(zapcore.DebugLevel))
+	case "dev":
+		opts = append(opts, kzap.UseDevMode(true), kzap.Level(zapcore.DebugLevel))
+	}
+	zlog := kzap.NewRaw(opts...).Sugar()
+	logf.SetLogger(zapr.NewLogger(zlog.Desugar()))
+	hostinfo.SetApp(kubetypes.AppProxy) // TODO(tomhjp): Advertise auth/noauth as well?
```

Comment:
> // TODO(tomhjp): Advertise auth/noauth as well?

We've historically treated 'app type' as somewhat immutable device type, not as something that changes as users update some config knobs, so I think it makes sense for it to be more generic.

We _might_ want to make it more specific than 'k8s-proxy' though. 'k8s-proxy' makes sense if we assume that in future this could be any proxy type (ingress, egress, api server proxy) and there is no way to distinguish between those